source/extensions/quic_listeners/quiche/platform/string_utils.cc

```diff
@@ -0,0 +1,96 @@
+#include "extensions/quic_listeners/quiche/platform/string_utils.h"
+
+// NOLINT(namespace-envoy)
+//
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <arpa/inet.h>
+#include <cstring>
+#include <string>
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/escaping.h"
+#include "absl/strings/str_format.h"
+#include "common/common/assert.h"
+#include "common/common/base64.h"
+
+namespace quiche {
+
+void Base64Encode(const uint8_t* data, size_t data_len, std::string* output) {
+  *output =
+      Envoy::Base64::encode(reinterpret_cast<const char*>(data), data_len, /*add_padding=*/false);
+}
+
+std::string HexDump(absl::string_view data) {
```

Comment:
Can you reuse https://github.com/envoyproxy/envoy/blob/master/source/common/common/hex.cc and https://github.com/envoyproxy/envoy/blob/master/source/common/common/hex.h here?

Reply:
They can not be used since they are not hexdump, rather they are equivalent to absl's HexStringToBytes and BytesToHexString functions.

I could move the body of HexDump to common/hex.cc and use it from here, the caveat is that the format of its output must not change, because quiche unit test expects the same output format([example](https://quiche.googlesource.com/quiche/+/refs/heads/master/quic/platform/api/quic_text_utils_test.cc#155)) from all platforms' implementations. Do you prefer to do it that way?

Reply:
Are these only test functions? I don't mind adding them if they are _only_ used for this, but should then have some comments or change their name. I think a few things:
* It's helpful to have these kinds of utils, so it would be nice to have them around in Envoy.
* It would ideally have a bunch of tests to ensure they behave as intended
*I'd prefer we did this as much as possible as pure C++, to avoid any of the evil of C string and buffer management.

Reply:
Thanks for the offline chat, Harvey. I have
1) Changed the visibility of "string_utils_lib" to private.
2) Opened https://github.com/envoyproxy/envoy/issues/6319 to cleanup the visibility of other quiche build rules.